pyout.py

```diff
@@ -43,19 +40,39 @@ class attribute.
     Attributes
     ----------
     term : blessings.Terminal instance
+
+    Examples
+    --------
+
+    Create a `Tabular` instance for two output fields, "name" and
+    "status".
+
+    >>> out = Tabular(["name", "status"], style={"status": {"width": 5}})
+
+    The first field, "name", is taken as the unique ID.  The `style`
+    argument is used to override the default width for the "status"
+    field that is defined by the class attribute `default_style`.
+
+    Write a row to stdout:
+
+    >>> out({"name": "foo", "status": "OK"})
+
+    Write another row, overriding the style:
+
+    >>> out({"name": "bar", "status": "BAD"},
+    ...     style={"status": {"color": "red", "bold": True}})
     """
 
     # TODO: Support things like auto-width, value-based coloring, etc.
```

Comment:
+1 ;)  for "auto-width" we might want to be able to say "minimal-width to be 10, maximal width 20, if exceeds - truncate on the right (or on the left, or in the middle, probably replaacing with ...)... or may be even make it scroll within those 20 (back and forth... I think it was done in some progress bar library that way). Needless to say that some options would not be available if output should be rendered once (e.g. when piping to the file)

Reply:
for value-based coloring -- may be it should be generalized into "value-based styling"? (e.g. making things bold when reaching some threshold) might generalize implementation etc

Reply:
Noted in the issue that really needs to be split up: #1